site/docs/backward-compatibility.md

```diff
@@ -0,0 +1,82 @@
+---
+layout: documentation
+title: Backward Compatibility
+---
+
+# Backward Compatibility
+
+Bazel is evolving, and we will make changes to Bazel that at times will be
+incompatible and require some changes from Bazel users.
+
+## At a glance
+
+1. Every breaking change is guarded with an `--incompatible_*` flag.
+1. Newly introduced incompatible flags default to off.
+1. For every `--incompatible_*` flag we have a github issue explaining
+   the change in behavior and giving a migration recipe.
+1. The migration window is at least one release long and is set by the author of the incompatible change.
+1. We announce what set of flags we flip with the next release one release in advance.
+1. APIs and behavior guarded by an `--experimental_*` flag can change at any time.
+1. Users should never run their production builds with `--experimental_*`  or `--incompatible_*` flags.
```

Comment:
I don't think we should be so strict. Users can do what they want at their own risk (which is the item above).

Reply:
Maybe "We do not recommend running their production builds"?

Reply:
We have to make it very clear that especially the owners of things that are dependent upon widely (such as rulesets) _should not_ expect any experimental or incompatible flags to be present.

I'd rather keep this strong language here.